smoke-tests/src/database/java/com/google/firebase/testing/database/DatabaseTest.java

```diff
@@ -40,6 +40,29 @@
 
   @Rule public final ActivityTestRule<Activity> activity = new ActivityTestRule<>(Activity.class);
 
+  @Test
+  public void setValueShouldFailWithPermissionDenied() throws Exception {
+    FirebaseAuth auth = FirebaseAuth.getInstance();
+    FirebaseDatabase database = FirebaseDatabase.getInstance();
+
+    auth.signOut();
+    Thread.sleep(1000); // TODO(allisonbm92): Introduce a better way to reduce flakiness.
```

Comment:
This is waiting for sign out to complete?

Reply:
Yes. Unfortunately, signOut doesn't return a Task. If sleeping here proves to be a problem, I think I might be able to work around it, but so far, I haven't seen any flakiness with the sleep.